docs/components/authentication/sign-up.mdx

```diff
@@ -18,64 +12,87 @@ import { CodeBlockTabs } from "@/components/CodeBlockTabs";
+The `<SignUp />` component renders a UI for signing up users. The functionality of the `<SignUp />` component is controlled by the instance settings you specify in your [Clerk Dashboard](https://dashboard.clerk.dev). You can further customize your `<SignUp />` component by passing additional properties at the time of rendering.
 
 ## Usage
 
 Below is basic implementation of the `<SignUp />` component. You can use this as a starting point for your own implementation
 
-<CodeBlockTabs options={['Next.js', 'React', 'Remix', 'Gatsby', 'JavaScript']}>
-
-```jsx copy filename="/pages/sign-up/[[...index]].[jsx/tsx]"
-import { SignUp } from "@clerk/nextjs";
-
-const SignUpPage = () => (
-  <SignUp path="/sign-up" routing="path" signInUrl="/sign-in" />
-);
-export default SignUpPage;
-```
-
-```jsx copy filename="sign-up.jsx"
-import { SignUp } from "@clerk/clerk-react";
-
-function SignUpPage() {
-  return <SignUp />;
-}
-```
-
-```jsx copy filename="app/routes/sign-up/$.tsx"
-import { SignUp } from "@clerk/remix";
-
-export default function SignUpPage() {
-  return (
-    <div style={{ border: "2px solid blue", padding: "2rem" }}>
-      <h1>Sign Up route</h1>
-      <SignUp routing={"path"} path={"/sign-up"} />
-    </div>
-  );
-}
-```
-
-```jsx copy filename="/pages/sign-up.js"
-import { SignUp } from "gatsby-plugin-clerk";
-
-export default function SignUpPage() {
-  return (
-    <div style={{ border: "2px solid blue", padding: "2rem" }}>
-      <h1>Sign Up route</h1>
-      <SignUp routing={"path"} path={"/sign-up"} />
-    </div>
-  );
-}
-```
-
-```js copy filename="sign-up.js"
-window.Clerk.mountSignUp(
-  document.getElementById("sign-up")
-);
-window.Clerk.openSignUp();
-```
-</CodeBlockTabs>
+<Tabs items={['Next.js', 'React', 'Remix', 'Gatsby', 'JavaScript']}>
+  <Tab>
+    You can embed the `<SignUp />` component using the [Next.js optional catch all route](https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes#optional-catch-all-routes). This allows you to redirect the user inside your application. The `<SignUp />` component should be mounted on a public page.
+
+    <Callout>The example below shows the Sign In page mounted on the url /sign-in. For more information on how to implement this, check out our [`<SignIn />` UI page](/docs/components/authentication/sign-in).</Callout>
+
+    <CodeBlockTabs options={["Pages router", "App router"]}>
+      ```jsx copy filename="/pages/sign-up/[[...index]].[jsx/tsx]"
+      import { SignUp } from "@clerk/nextjs";
+
+      const SignUpPage = () => (
+        <SignUp path="/sign-up" routing="path" signInUrl="/sign-in" />
+      );
+      export default SignUpPage;
+      ```
+
+      ```jsx copy filename="/app/sign-up/[[...sign-up]]/page.[jsx/tsx]"
+      import { SignUp } from "@clerk/nextjs";
+
+      export default function Page() {
+        return <SignUp />;
+      }
+      ```
+    </CodeBlockTabs>
```

Comment:
```suggestion
    <CodeBlockTabs options={["App Router", "Pages Router"]}>
      ```jsx copy filename="/app/sign-up/[[...sign-up]]/page.[jsx/tsx]"
      import { SignUp } from "@clerk/nextjs";

      export default function Page() {
        return <SignUp />;
      }
      ```

      ```jsx copy filename="/pages/sign-up/[[...index]].[jsx/tsx]"
      import { SignUp } from "@clerk/nextjs";

      const SignUpPage = () => (
        <SignUp path="/sign-up" routing="path" signInUrl="/sign-in" />
      );
      export default SignUpPage;
      ```
    </CodeBlockTabs>
```